File: root/common/src/main/java/org/sc/common/rest/controller/TrailPreviewResponse.java
package org.sc.common.rest.controller;

import java.util.Collections;
import java.util.List;
import java.util.Set;

public class TrailPreviewResponse extends RESTResponse {
    final List<TrailPreview> trailPreviews;

    public TrailPreviewResponse(final List<TrailPreview> trailPreviews,
                                final Status status,
                                final Set<String> messages) {
        super(status, messages);
        this.trailPreviews = trailPreviews;
    }

    public TrailPreviewResponse(List<TrailPreview> trailPreviews) {
        super(Status.OK, Collections.emptySet());
        this.trailPreviews = trailPreviews;
    }

    public List<TrailPreview> getTrailPreviews() {
        return trailPreviews;
    }
}

File: root/common/src/main/java/org/sc/common/rest/response/TrailPreviewResponse.kt
package org.sc.common.rest.response

import org.sc.common.rest.Status
import org.sc.common.rest.TrailPreviewDto

data class TrailPreviewResponse (val status: Status,
                                 val messages: Set<String>,
                                 val content: List<TrailPreviewDto>) :
    RESTResponse()